components/list/list-item-role-mixin.js

```diff
@@ -19,8 +19,9 @@ export const ListItemRoleMixin = superclass => class extends superclass {
 
 		const separators = parent.getAttribute('separators');
 
-		this.role = parent.grid ? 'rowgroup' : 'listitem';
+		this.role = parent.hasAttribute('grid') ? 'rowgroup' : 'listitem';
```

Comment:
This was actually randomly broken it seemed.  Like the `grid` property wasn't always set on the parent in time.